src/JustSaying/AwsTools/MessageHandling/SnsMessagePublisher.cs

```diff
@@ -93,26 +97,54 @@ public async Task PublishAsync(Message message, PublishMetadata metadata, Cancel
     private PublishRequest BuildPublishRequest(Message message, PublishMetadata metadata)
     {
         var messageToSend = _serializationRegister.Serialize(message, serializeForSnsPublishing: true);
+
+        string contentEncoding = null;
+        if (CompressionOptions?.CompressionEncoding is { } compressionEncoding && CompressionRegistry is not null)
+        {
+            var bodyByteLength = Encoding.UTF8.GetByteCount(messageToSend); // We should probably also include the length of the message attributes
+            if (bodyByteLength > (CompressionOptions?.MessageLengthThreshold ?? int.MaxValue)) // Well under 256KB
```

Comment:
This took me a minute to grok - maybe refactor to be something like `if (CompressionOptions?.MessageLengthThreshold is { } threshold && bodyByteLength > threshold) { ... }`?